### PR TITLE
Python output class

### PR DESCRIPTION
**Pull request by Thoralf-M**
# Description of change

Add Output class and update classes that are used inside so they can be easily deserialized

## Links to any relevant issues

#42 

## Type of change

- Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How the change has been tested

Added tests, running examples

## Change checklist

- [ ] I have followed the contribution guidelines for this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that new and existing unit tests pass locally with my changes

### CODE REVIEW

File: bindings/python/iota_sdk/types/token_scheme.py [hunk 1]
```diff
@@ -1,30 +1,25 @@
...
+    mintedTokens: HexStr
+    meltedTokens: HexStr
+    maximumSupply: HexStr
+    type: int = 0
 
     def as_dict(self):
         config = dict(self.__dict__)
         
-        if 'mintedTokens' in config:
+        if isinstance(config['mintedTokens'], int):
```
Comment (Alex6323) [inline bindings/python/iota_sdk/types/token_scheme.py:18]: Now I know that Pythoneans are crazy if they like a name like: `isinstance`. You must be kidding me!!!

File: bindings/python/examples/client/submit_and_read_block.py [hunk 2]
```diff
@@ -80,9 +80,9 @@
 
 # Get the whole block
 block = client.get_block_data(block_id)
-payload_out = block['payload']
-tag_hex_out = block['payload']['tag']
-message_hex_out = block['payload']['data']
+payload_out = block.payload
+tag_hex_out = block.payload['tag']
+message_hex_out = block.payload['data']
```
Comment (Alex6323) [inline bindings/python/examples/client/submit_and_read_block.py:85]: rename to `data_hex_out`?

File: bindings/python/iota_sdk/client/_high_level_api.py [hunk 3]
```diff
@@ -44,20 +47,24 @@ def as_dict(self):
...
         """Try to get OutputResponse from provided OutputIds.
            Requests are sent in parallel and errors are ignored, can be useful for spent outputs.
         """
-        return self._call_method('getOutputsIgnoreErrors', {
-            'outputIds': output_ids
+        result =  self._call_method('getOutputsIgnoreErrors', {
+            'outputIds': list(map(lambda o: o.output_id, output_ids))
         })
+        outputs = [from_dict(OutputWithMetadata, o) for o in result]
+        return outputs
```
Comment (Alex6323) [inline bindings/python/iota_sdk/client/_high_level_api.py]: Can we rename `result` to `outputs`, and then just
```py
return [from_dict(OutputWithMetadata, o) for o in outputs]
```
?

File: bindings/python/iota_sdk/client/_high_level_api.py [hunk 4]
```diff
@@ -44,20 +47,24 @@ def as_dict(self):
...
-    def get_outputs(self, output_ids: List[OutputId]):
+    def get_outputs(self, output_ids: List[OutputId]) -> List[OutputWithMetadata]:
         """Fetch OutputResponse from provided OutputIds (requests are sent in parallel).
         """
-        return self._call_method('getOutputs', {
-            'outputIds': output_ids
+        result =  self._call_method('getOutputs', {
+            'outputIds': list(map(lambda o: o.output_id, output_ids))
         })
+        outputs = [from_dict(OutputWithMetadata, o) for o in result]
```
Comment (kwek20) [inline bindings/python/iota_sdk/client/_high_level_api.py]: i think this line is by far my fav in python

File: bindings/python/examples/client/submit_and_read_block.py (hunk 2 above)
Comment (Thoralf-M) [reply bindings/python/examples/client/submit_and_read_block.py:85]: But it's about the message defined in L34

File: bindings/python/iota_sdk/client/_high_level_api.py (hunk 3 above)
Comment (Thoralf-M) [reply bindings/python/iota_sdk/client/_high_level_api.py]: Yes we can!

Comment (Thoralf-M) [reply bindings/python/iota_sdk/client/_high_level_api.py]: Updated

File: bindings/python/examples/client/04_get_output.py [hunk 5]
```diff
@@ -12,5 +12,5 @@
 client = Client(nodes=[node_url])
 
 # Get an outputs by its id
-output = client.get_output('0x1e857d380f813d8035e487b6dfd2ff4740b6775273ba1b576f01381ba2a1a44c0000')
-print(json.dumps(output, indent=4))
+output_metadata = client.get_output('0x022aefa73dff09b35b21ab5493412b0d354ad07a970a12b71e8087c6f3a7b8660000')
```
Comment (DaughterOfMars) [inline bindings/python/examples/client/04_get_output.py]: Why would this be called metadata if it's `get_output`?

**Review by DaughterOfMars — DISMISSED** [review]

File: bindings/python/examples/client/04_get_output.py (hunk 5 above)
Comment (Thoralf-M) [reply bindings/python/examples/client/04_get_output.py]: Well that's what we also return in Rust `get_output(&self, output_id: &OutputId) -> Result<OutputWithMetadata>`

Comment (Thoralf-M) [reply bindings/python/examples/client/04_get_output.py]: Actually missed the "with"

Comment (Thoralf-M) [reply bindings/python/examples/client/04_get_output.py]: Renamed

Comment (thibault-martinez) [reply bindings/python/examples/client/04_get_output.py]: FYI this has been changed in 2.0, output just returns output

**Review by DaughterOfMars — DISMISSED** [review]

File: bindings/python/iota_sdk/types/token_scheme.py (hunk 1 above)
Comment (Alex6323) [reply bindings/python/iota_sdk/types/token_scheme.py:18]: no response, I take this as a confirmation :trollface:

File: bindings/python/examples/client/submit_and_read_block.py (hunk 2 above)
Comment (Alex6323) [reply bindings/python/examples/client/submit_and_read_block.py:85]: I have to work with the context Gh is providing me! :trollface:

File: bindings/python/iota_sdk/types/unlock_condition.py [hunk 6]
```diff
@@ -79,7 +61,7 @@ def __init__(self, amount, return_address):
             Return address
         """
         super().__init__(type=UnlockConditionType.StorageDepositReturn,
-                         amount=amount, return_address=return_address)
+                         amount=str(amount), returnAddress=return_address)
```
Comment (Alex6323) [inline bindings/python/iota_sdk/types/unlock_condition.py:64]: I am almost afraid to ask: we can't unify the casing, right? right??

**Review by Alex6323 — DISMISSED** [review]

File: bindings/python/iota_sdk/types/unlock_condition.py (hunk 6 above)
Comment (Thoralf-M) [reply bindings/python/iota_sdk/types/unlock_condition.py:64]: Well parameters are usually snake_case in Python, but we serialize/deserialize the keys with camelCase which isn't that simply as in Rust with serde_json :cry: Aat least I haven't seen a simple way to do that, happy if someone suggests something that does it better :trollface:

File: bindings/python/iota_sdk/client/_high_level_api.py [hunk 7]
```diff
@@ -44,29 +46,30 @@ def as_dict(self):
 
 class HighLevelAPI():
 
-    def get_outputs(self, output_ids: List[OutputId]):
+    def get_outputs(self, output_ids: List[OutputId]) -> List[OutputWithMetadata]:
         """Fetch OutputResponse from provided OutputIds (requests are sent in parallel).
```
Comment (kwek20) [inline bindings/python/iota_sdk/client/_high_level_api.py]: OutputResponse is mentioned here and below, even though i dont think they exist in python. should rename it to idk.. "fetch OutputWithMetadata"

File: bindings/python/iota_sdk/types/address.py [hunk 8]
```diff
@@ -2,42 +2,26 @@
...
-            The type of the Address
-        address_or_id : string
-            The address to use. Can either be an hex encoded ED25519 address or NFT/Alias id
-        """
-        self.type = type
-        self.address_or_id = address_or_id
+    type: int
+    pubKeyHash: Optional[HexStr] = None
+    aliasId: Optional[HexStr] = None
+    nftId: Optional[HexStr] = None
```
Comment (kwek20) [inline bindings/python/iota_sdk/types/address.py:21]: having this here instead of in their respective classes is just to make the as_dict easier right?

Comment (kwek20) [reply bindings/python/iota_sdk/types/address.py:21]: same for feature, i guess.

File: bindings/python/iota_sdk/types/output.py [hunk 9]
```diff
@@ -68,3 +87,11 @@ def from_dict(cls, dict: Dict) -> OutputWithMetadata:
         for k, v in dict.items():
             setattr(obj, k, v)
         return obj
+
+    def as_dict(self):
+        config = dict()
+
+        config['metadata'] = self.metadata.__dict__
```
Comment (kwek20) [inline bindings/python/iota_sdk/types/output.py:103]: this means metadata still needs to be done right?

File: bindings/python/iota_sdk/types/unlock_condition.py (hunk 6 above)
Comment (kwek20) [reply bindings/python/iota_sdk/types/unlock_condition.py:64]: lets change it so we just do normal rust in bindings and python, and snake_case internally in js. thatll eb a fun pr :D

File: bindings/python/iota_sdk/types/output.py
Comment (kwek20) [inline bindings/python/iota_sdk/types/output.py:1]: Output class types will be a different pr?

File: bindings/python/iota_sdk/types/address.py (hunk 8 above)
Comment (Thoralf-M) [reply bindings/python/iota_sdk/types/address.py:21]: Also for the deserialization
Doesn't really look good, but it works and I don't know how else to do it better

File: bindings/python/iota_sdk/types/output.py (hunk 9 above)
Comment (Thoralf-M) [reply bindings/python/iota_sdk/types/output.py:103]: No, but metadata has only "simple" fields, so __dict__ does the job and no extra method is needed

File: bindings/python/iota_sdk/types/output.py
Comment (Thoralf-M) [reply bindings/python/iota_sdk/types/output.py:1]: It's so small, added it now

File: bindings/python/iota_sdk/client/_high_level_api.py (hunk 7 above)
Comment (Thoralf-M) [reply bindings/python/iota_sdk/client/_high_level_api.py]: Done

File: bindings/python/iota_sdk/types/output.py
Comment (DaughterOfMars) [reply bindings/python/iota_sdk/types/output.py:1]: Should it be used in the Output?

**Review by Alex6323 — APPROVED** [review]

File: bindings/python/iota_sdk/types/output.py
Comment (Thoralf-M) [reply bindings/python/iota_sdk/types/output.py:1]: Unfortunately doesn't work with deserialization `WrongTypeError: wrong value type for field "type" - should be "OutputType" instead of value "3" of type "int"`

**Review by kwek20 — APPROVED** [review]
